pkgs/games/minecraft-server/derivation.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchurl, nixosTests, jre_headless, version, url, sha1 }:
+stdenv.mkDerivation {
+  pname = "minecraft-server";
+  inherit version;
+
+  src = fetchurl { inherit url sha1; };
+
+  preferLocalBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin $out/lib/minecraft
+    cp -v $src $out/lib/minecraft/server.jar
+
+    cat > $out/bin/minecraft-server << EOF
+    #!/bin/sh
+    exec ${jre_headless}/bin/java \$@ -jar $out/lib/minecraft/server.jar nogui
```

Comment:
What version is the jre? Newer versions run best with the latest, but anything older than 1.12 (if I remember correctly) needs 1.8 to run well. 
I've had some bugs client side before, because I was using version 16 of java with 1.8

Reply:
Yep, the API does provide some info on this:
```json
  "id": "1.16.5",
  "javaVersion": {
    "component": "jre-legacy",
    "majorVersion": 8
  }
```
I'll fix this now.